common/scripts/access-control/flexible-server/Grant-FlexibleServerAccess-NotifyDB-Reader.ps1

```diff
@@ -0,0 +1,109 @@
+[string]$PlatformMISubscriptionId = $env:PLATFORM_MI_SUBSCRIPTION_ID 
+[string]$SubscriptionName = $env:SUBSCRIPTION_NAME
+[string]$WorkingDirectory = $PWD
+[string]$PostgresReaderAdGroup = $env:PG_READER_AD_GROUP
+[string]$PostgresWriterAdGroup = $env:PG_WRITER_AD_GROUP
+
+[string]$functionName = $MyInvocation.MyCommand
+[DateTime]$startTime = [DateTime]::UtcNow
+[int]$exitCode = -1
+[bool]$setHostExitCode = (Test-Path -Path ENV:TF_BUILD) -and ($ENV:TF_BUILD -eq "true")
+[bool]$enableDebug = (Test-Path -Path ENV:SYSTEM_DEBUG) -and ($ENV:SYSTEM_DEBUG -eq "true")
+
+Set-Variable -Name ErrorActionPreference -Value Continue -scope global
+Set-Variable -Name VerbosePreference -Value Continue -Scope global
+
+if ($enableDebug) {
+    Set-Variable -Name DebugPreference -Value Continue -Scope global
+    Set-Variable -Name InformationPreference -Value Continue -Scope global
+}
+
+Write-Host "${functionName} started at $($startTime.ToString('u'))"
+Write-Debug "${functionName}:PostgresHost:$PostgresHost"
+Write-Debug "${functionName}:PostgresDatabase:$PostgresDatabase"
+Write-Debug "${functionName}:SubscriptionName=$SubscriptionName"
+Write-Debug "${functionName}:WorkingDirectory=$WorkingDirectory"
+Write-Debug "${functionName}:PostgresReaderAdGroup=$PostgresReaderAdGroup"
+Write-Debug "${functionName}:PostgresWriterAdGroup=$PostgresWriterAdGroup"
+
+[System.IO.DirectoryInfo]$scriptDir = $PSCommandPath | Split-Path -Parent
+Write-Debug "${functionName}:scriptDir.FullName:$($scriptDir.FullName)"
+
+function Get-SQLScriptToGrantReadPermissions {
+    [System.Text.StringBuilder]$builder = [System.Text.StringBuilder]::new()
+    [void]$builder.Append("GRANT SELECT ON ALL TABLES IN SCHEMA public TO `"$PostgresReaderAdGroup`";")
+    [void]$builder.Append("GRANT SELECT ON ALL SEQUENCES IN SCHEMA public TO `"$PostgresReaderAdGroup`";")
+    [void]$builder.Append("REVOKE EXECUTE ON ALL FUNCTIONS IN SCHEMA public FROM `"$PostgresReaderAdGroup`";")
+    [void]$builder.Append("REVOKE EXECUTE ON ALL PROCEDURES IN SCHEMA public FROM `"$PostgresReaderAdGroup`";")
+    [void]$builder.Append("ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT ON TABLES TO `"$PostgresReaderAdGroup`";")
+    [void]$builder.Append("ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT ON SEQUENCES TO `"$PostgresReaderAdGroup`";")
+    return $builder.ToString()
+}
+
+[string]$tempFolder=[System.IO.Path]::GetTempPath()
+[System.IO.FileInfo]$assignReadPermissionsTempFile = $tempFolder + $(New-Guid).ToString() + ".sql";
+
+try {
+    [System.IO.DirectoryInfo]$moduleDir = Join-Path -Path $WorkingDirectory -ChildPath "common/scripts/modules/psql"
+    Write-Debug "${functionName}:moduleDir.FullName=$($moduleDir.FullName)"
+    Import-Module $moduleDir.FullName -Force
+
+    Write-Host "Connecting to Azure..."
+    $azureContext = (Connect-AzAccount -Identity -AccountId $PlatformMIClientId -Tenant $PlatformMITenantId -Subscription $PlatformMISubscriptionId).context
```

Comment:
@kbabigumira - Should this now be connecting with the 'SSVADPDEVMI3401-flux-notify-api' managed identity to assign the reader permissions?

Reply:
As discussed, I have renamed it to NotificationApiMIClientId to make it clearer